partner_compassion/model/account_banking_mandate_compassion.py

```diff
@@ -0,0 +1,111 @@
+# -*- encoding: utf-8 -*-
+##############################################################################
+#
+#    Copyright (C) 2014-2015 Compassion CH (http://www.compassion.ch)
+#    Releasing children from poverty in Jesus' name
+#    @author: Steve Ferry
+#
+#    The licence is in the file __openerp__.py
+#
+##############################################################################
+from openerp.osv import orm
+
+
+class Account_Banking_Mandate(orm.Model):
+    """ This class upgrade the partners.bank to match Compassion needs.
+    """
+    _inherit = 'account.banking.mandate'
+
+    def create(self, cr, uid, data, context):
+        """Override function to notify creation in a message on partner feed
+        """
+        obj_bank = self.pool['res.partner.bank']
+        account = obj_bank.browse(cr, uid, data['partner_bank_id'],
+                                  context=context)
+
+        account.partner_id.pool.get('mail.thread') \
+            .message_post(cr, uid, account.partner_id.id,
+                          "For account: " +
+                          account.acc_number,
+                          "New mandate created",
+                          'comment',
+                          context={'thread_model':
+                                   account.partner_id._name})
+
+        result = super(Account_Banking_Mandate, self).create(cr, uid, data,
+                                                             context=context)
+        return result
+
+    def validate(self, cr, uid, ids, context=None):
+        """Override function to notify validation in a message on partner feed
+        """
+
+        obj_mandate = self.pool['account.banking.mandate']
+
+        for mandate in obj_mandate.browse(cr, uid, ids, context=context):
+            mandate.partner_id.pool.get('mail.thread') \

```

Comment:
Better to have a variable declared before the loop :  `thread_obj = self.pool.get('mail.thread')`